tensorpipe/core/pipe.cc

```diff
@@ -53,6 +53,7 @@ struct ReadOperation {
   };
   std::vector<Payload> payloads;
   struct Tensor {
+    DeviceType deviceType;
```

Comment:
Why not make two `Tensor` structs and store them in a `tensors` and a `cudaTensors` fields?